draft/0004-user-sign-in-experience.md

```diff
@@ -0,0 +1,669 @@
+- **Verification ID**: A unique identifier for a verification record. It's a random string generated by the Logto server when a verification record is initiated. Users need to provide the verification ID along with additional security information to verify their identity. The verification ID can also be used to track the status of the verification record within the interaction.
+- **Multi-factor authentication (MFA)**: A security process that requires more than one verification factor to complete the interaction. It can be used as an additional security check during the sign-in experience.
+- **First party application**: A application that is built and managed by the same company that uses Logto for user authentication.
+- **Third party application**: A application that is owned and managed by a different company that uses Logto as a external identity provider. E.g. A external application that uses Logto as a enterprise SSO provider.
+
+### 3.2 User experience
+
+The user sign-in experience in Logto consists of the following key steps:
+
+1. **Sign-in experience initiation**: When the client application sends an authentication request to Logto, the user is redirected to Logto's experience app, initiating a new interaction for the authentication process.
+2. **Identification**: The user must provide an verified identifier to Logto, such as a username, email address, phone number, social identity, or SSO identity. This identifier is used to recognize the user during the interaction.
+3. **Verification**: The process to verify a user's identity by requesting additional security information. This can be a one-time code, or a passkey, etc. Upon the security settings, multiple verification methods can be used.
+4. **Profile fulfillment**: During the interaction, the user may be prompted to complete their profile and MFA settings. This can be necessary for user registration or mandatory updates for existing users' profiles and MFA settings.
+5. **Authentication**: Once the user has completed all the necessary steps, they can submit the interaction to get authenticated by Logto. All the user's profile and MFA settings updates will be saved to the user database.
+6. **Consent**: Users may need to grant consent for specific actions upon the requested authorization details. This could involve sharing personal information, allowing access for their organization, or granting permissions to specific resources. For first-party applications, consent is implicit and handled automatically by Logto once user get authenticated. For third-party applications, consent is explicit, and the user will be prompted to grant the necessary permissions after a successful interaction submission.
+7. **Authorization**: The final step in the sign-in experience. Once the user has been authenticated and consented to the necessary actions, they will be redirected back to the client application with granted authorization.
```

Comment:
It might be confused with the interaction submission. One way I can think of is replacing the `Authentication` step with `Submission`. But not sure about that either. Authentication might be more precise.